### ISSUE TITLE
Question about diffusion prior 

### ISSUE THREAD

**Issue by xiaotingxuan** [lucidrains/DALLE2-pytorch#265]
Hi，In my project, I only train the diffusion prior network. 

I use train_prior_config.example.json which is provided in [here](https://github.com/lucidrains/DALLE2-pytorch/blob/main/configs/train_prior_config.example.json)
I only change the hyperparameter "use_ema" to false. And I use MScoco dataset(karpathy split) to train . During training, the loss type is "mse loss", and the train_Ioss can go down to 0.2.

For inference, I use diffusion_prior.sample(tokenized_text, n_samples_per_batch=2, cond_scale=1.0) to generate CLIP image embedding

I think the generated CLIP image embedding should be similar with the ground-truth CLIP image embedding. I find the the cosine similarity between them are about 0.7.  Is this normal?

**Comment by Lixin-Liu** [comment]
Hi, I have a question. How long did it take you to train the model?

**Comment by xiaotingxuan** [comment]
About 70 minutes for one epoch，use one NVIDIA TITAN RTX

**Comment by Lixin-Liu** [comment]
Thank you!

**Comment by vae1207** [comment]
How do you process the dataset？I don't know why my loss value can't go down, Looks so weird

**Comment by cxhermagic** [comment]
I want to train Diffusion Prior model , how to get the PriorEmbeddingDataset? Thank you.

**Comment by xiaotingxuan** [comment]
If you mean get clip text embedding and clip image embedding for train prior，I think you can read [ClipCap's code](https://github.com/rmokady/CLIP_prefix_caption/blob/main/parse_coco.py) for reference. Extract CLIP features, then create your  PriorEmbeddingDataset
